src/StoreApi/SessionHandler.php

```diff
@@ -110,4 +109,39 @@ public function save_data() {
 			$this->_dirty = false;
 		}
 	}
+
+	/**
+	 * Since there is no interface for the abstract session class, we need to wrap calls for methods present on
+	 * WC_Session_Handler class that do not exist here (mostly cookie related methods) to prevent fatal errors
+	 * due to an optimistic approach to calling methods on the session object.
+	 * An example of this would be https://github.com/woocommerce/woocommerce-blocks/issues/9116
+	 *
+	 * @param string $name Method name.
+	 * @param array  $arguments Method arguments.
+	 * @throws \Error If the method does not exist on WC_Session_Handler.
+	 *
+	 * @return void
+	 */
+	public static function __callStatic( $name, $arguments ) {
```

Comment:
Instead of this, can we not just make a `init_session_cookie` method to extend the core class properly? We can document inline and make it a wrapper for `init_session_from_token`.

Reply:
Then if a rename make sense, we can do so upstream.

Reply:
I don't think that is the only method [we're going to have trouble with](https://github.com/woocommerce/woocommerce-blocks/pull/5953#issuecomment-1551198659). Plus have a cookie named method on a non cookie based session class.. er..
On Core, cookie session handler should be extending an agnostic class or implementing an interface that we could use. We can move this bad design into a compatibility class to ensure backwards compatibility.